app/Providers/RouteServiceProvider.php

```diff
@@ -33,13 +39,19 @@ public function boot(Router $router)
     /**
      * Define the routes for the application.
      *
-     * @param \Illuminate\Routing\Router $router router
+     * @param Router  $router  router
+     * @param Request $request router
      *
      * @return void
      */
-    public function map(Router $router)
+    public function map(Router $router, Request $request)
     {
-        $router->group(['namespace' => $this->namespace], function ($router) {
+        if ($request->segment(1) == self::ADMIN_PATH) {
```

Comment:
it's not good to check `admin` in URL and add namespace. You have to add namespaces for both `Admin` and `Frontend`